pkg/node/node.go

```diff
@@ -1665,89 +1638,509 @@ func (s *Service) buildInitiatorsArray(initiators []string, arrayID string) []go
+		return s.isRemoteToOtherArray(ctx, arr, remoteArr)
+	}
+
+	registerHostFunc = func(s *Service, ctx context.Context, client gopowerstore.Client, arrayID string, initiators []string, connType gopowerstore.HostConnectivityEnum) error {
+		return s.registerHost(ctx, client, arrayID, initiators, connType)
+	}
+)
+
 // register host
-func (s *Service) createHost(ctx context.Context, initiators []string, client gopowerstore.Client, arrayID string) (id string, err error) {
-	osType := gopowerstore.OSTypeEnumLinux
-	reqInitiators := s.buildInitiatorsArray(initiators, arrayID)
-	description := fmt.Sprintf("k8s node: %s", s.opts.KubeNodeName)
+func (s *Service) createHost(
+	ctx context.Context,
+	initiators []string,
+) (string, error) {
+	nodeLabels, err := k8sutils.GetNodeLabels(context.Background(), s.opts.KubeConfigPath, s.opts.KubeNodeName)
+	if err != nil {
+		return "", fmt.Errorf("failed to get node labels for node %s: %v", s.opts.KubeNodeName, err)
+	}
+	var primaryArrayID string
 
-	var createParams gopowerstore.HostCreate
-	defaultHeaders := client.GetCustomHTTPHeaders()
-	if defaultHeaders == nil {
-		defaultHeaders = make(http.Header)
+	// Step 1: Check if this node matches at least one labeled Metro array
+	anyLabelMatch := false
+	for _, arr := range getArrayfn(s) {
+		if strings.ToLower(arr.MetroTopology) == "uniform" && len(arr.Labels) == 1 {
+			if labelsMatch(arr.Labels, nodeLabels) {
+				anyLabelMatch = true
+				break
+			}
+		}
 	}
-	customHeaders := defaultHeaders
-	k8sMetadataSupported := common.IsK8sMetadataSupported(client)
-	if k8sMetadataSupported {
-		customHeaders.Add("DELL-VISIBILITY", "internal")
-		client.SetCustomHTTPHeaders(customHeaders)
 
-		if s.opts.KubeNodeName == "" {
-			log.Warnf("KubeNodeName value is not set")
-			createParams = gopowerstore.HostCreate{
-				Name: &s.nodeID, OsType: &osType, Initiators: &reqInitiators,
-				Description: &description,
+	for _, arr := range getArrayfn(s) {
+		log.Infof("[createHost] Processing array %s (%s)", arr.GlobalID, arr.IP)
+		// 1) Skip if already registered
+		if getIsHostAlreadyRegistered(s, ctx, arr.GetClient(), initiators) {
+			log.Infof("[createHost] Already registered on %s, skipping", arr.GlobalID)
+			if primaryArrayID == "" {
+				primaryArrayID = arr.GlobalID
+			}
+			continue
+		}
+
+		// 2) Metro vs Non‑Metro
+		if strings.ToLower(arr.MetroTopology) != "uniform" {
+			log.Infof("[createHost] Non‑Metro array %s → registering LocalOnly", arr.GlobalID)
+			if err := s.registerHost(
+				ctx, arr.GetClient(), arr.GlobalID, initiators,
+				gopowerstore.HostConnectivityEnumLocalOnly,
+			); err != nil {
+				return "", fmt.Errorf("failed LocalOnly on %s: %v", arr.GlobalID, err)
+			}
+			if primaryArrayID == "" {
+				primaryArrayID = arr.GlobalID
+			}
+			continue
+		}
+
+		if len(arr.Labels) > 1 {
+			log.Infof("[createHost] Skipping Metro array %s: more than one label", arr.GlobalID)
```

Comment:
Can we log at Warning level as this is a misconfiguration from users end, they should take care of this.